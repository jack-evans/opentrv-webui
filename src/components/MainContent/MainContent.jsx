import React, { Component } from 'react'
import './MainContent.scss'

class MainContent extends Component {
  render () {
    return (
      <div className='MainContent'>
        <p>
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    )
  }
}

export default MainContent
