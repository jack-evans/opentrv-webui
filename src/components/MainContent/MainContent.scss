
.MainContent {
  display: flex;
  top: 60px;
}