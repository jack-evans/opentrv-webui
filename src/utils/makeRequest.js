
export default function makeRequest (apiPath, options) {
  options.mode = 'cors'
  if (options.headers) {
    options.headers['Content-Type'] = 'application/json'
  } else {
    options.headers = {
      'Content-Type': 'application/json'
    }
  }

  return global.fetch(`http://localhost:3001${apiPath}`, options)
    .then(internal.checkStatus)
    .then(internal.parseJSON)
}

function checkStatus (response) {
  if (response.status >= 200 && response.status < 300) {
    return response
  } else {
    let error = new Error(response.statusText)
    error.response = response
    throw error
  }
}

function parseJSON (response) {
  return response.json()
}

export const internal = {
  checkStatus: checkStatus,
  parseJSON: parseJSON
}
