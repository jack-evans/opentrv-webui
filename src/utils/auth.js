import makeRequest from './makeRequest'
import decode from 'jwt-decode'

export function isAuthenticated () {
  const token = getToken()
  if (!token) {
    return false
  } else {
    const decodedToken = decode(token)
    const date = new Date(0)
    date.setUTCSeconds(decodedToken.exp)
    if (date < new Date()) {
      return false
    } else {
      return true
    }
  }
}

export function loginUser (user) {
  let url = '/api/v1/user/login'
  let options = {
    method: 'POST',
    json: true,
    headers: {
      'content-type': 'application/json'
    },
    body: JSON.stringify(user)
  }

  return makeRequest(url, options)
    .then(res => {
      setToken(res.token)
      return Promise.resolve()
    })
}

export function setToken (token) {
  window.localStorage.setItem('token', token)
}

export function getToken () {
  return window.localStorage.getItem('token')
}

export function logout () {
  window.localStorage.removeItem('token')
}
