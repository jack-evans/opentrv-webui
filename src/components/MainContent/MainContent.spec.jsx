import React from 'react'
import MainContent from './MainContent'
import { setAuthenticated } from '../../utils/auth'
import { MemoryRouter } from 'react-router-dom'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('MainContent.jsx', () => {
  it('renders without crashing', () => {
    expect(JSON.stringify(
      Object.assign({}, MainContent, { _reactInternalInstance: 'censored' })
    )).toMatchSnapshot()
  })

  describe('when the inital entry is "/"', () => {
    it('loads the Welcome page', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={['/']}>
          <MainContent />
        </MemoryRouter>
      )
      expect(wrapper.find('.Welcome')).toHaveLength(1)
    })
  })

  describe('when the inital entry is "/login"', () => {
    it('loads the Login page', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={['/login']}>
          <MainContent />
        </MemoryRouter>
      )
      expect(wrapper.find('.Login')).toHaveLength(1)
    })
  })

  // not ready for this test
  describe.skip('when the inital entry is "/register"', () => {
    it('loads the Register page', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={['/register']}>
          <MainContent />
        </MemoryRouter>
      )
      expect(wrapper.find('.Register')).toHaveLength(1)
    })
  })

  describe('when the initial entry is "/overview"', () => {
    let wrapper

    beforeEach(() => {
      setAuthenticated(true)
      wrapper = mount(
        <MemoryRouter initialEntries={['/overview']}>
          <MainContent />
        </MemoryRouter>
      )
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('loads the Overview page', () => {
      expect(wrapper.find('.Overview')).toHaveLength(1)
    })

    it('renders the Header component', () => {
      expect(wrapper.find('.Header')).toHaveLength(1)
    })

    it('renders the SideBarMenu', () => {
      expect(wrapper.find('.SideBarMenu')).toHaveLength(1)
    })

    describe.skip('when the menu button is clicked', () => {
      it('changes the state of sideBarMenuOpen', () => {
        const mainContent = wrapper.find('MainContent')
        console.log(mainContent.instance())
        // expect(mainContent.state().sideBarMenuOpen).toBe(false)
        wrapper.find('Header').simulate('click')
        expect(wrapper.state().sideBarMenuOpen).toBe(true)
      })

      describe('multiple times', () => {
        it('changes the state of sideBarMenuOpen', () => {
          expect(wrapper.state().sideBarMenuOpen).toBe(false)
          wrapper.find('Header').simulate('click')
          expect(wrapper.state().sideBarMenuOpen).toBe(true)
          wrapper.find('Header').simulate('click')
          expect(wrapper.state().sideBarMenuOpen).toBe(false)
          wrapper.find('Header').simulate('click')
          expect(wrapper.state().sideBarMenuOpen).toBe(true)
        })
      })
    })
  })

  describe('when the initial entry begins with "/devices"', () => {
    let wrapper

    beforeEach(() => {
      setAuthenticated(true)
      wrapper = mount(
        <MemoryRouter initialEntries={['/devices/1234']}>
          <MainContent />
        </MemoryRouter>
      )
    })

    afterEach(() => {
      wrapper.unmount()
    })

    it('loads the DevicePanel page', () => {
      expect(wrapper.find('DevicePanel')).toHaveLength(1)
    })

    it('renders the Header component', () => {
      expect(wrapper.find('.Header')).toHaveLength(1)
    })

    it('renders the SideBarMenu', () => {
      expect(wrapper.find('.SideBarMenu')).toHaveLength(1)
    })
  })

  describe('when the inital entry is "/random"', () => {
    it('loads the NotFound page', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={['/random']}>
          <MainContent />
        </MemoryRouter>
      )
      expect(wrapper.find('.NotFound')).toHaveLength(1)
    })
  })
})
