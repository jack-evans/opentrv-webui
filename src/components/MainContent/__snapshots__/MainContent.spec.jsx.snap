// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent.jsx renders without crashing 1`] = `"{\\"_reactInternalInstance\\":\\"censored\\"}"`;
